### PR TITLE
feat: 新增 Empty 组件

### PR DESCRIPTION
**Pull request by vimplus**
**What kind of change does this PR introduce?** (check at least one)

-   [ ] Bugfix
-   [x] Feature
-   [ ] Code style update
-   [ ] Refactor
-   [ ] Docs
-   [ ] Build-related changes
-   [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

-   [ ] Yes
-   [x] No

If yes, please describe the impact and migration path for existing applications:

**Related issue (if exists):**

**Other information:**

### CODE REVIEW

File: components/empty/props.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+     * @description image URL of empty
+     */
+    imageSrc: {
+        type: String,
+        default: '',
+    },
+    /**
+     * @description image style of empty
+     */
+    imageStyle: {
```
Comment (wanchun) [inline components/empty/props.ts:15]: style类型应该是[Object,Array,String] as PropType<StyleValue>。

StyleValue由Vue导出

File: components/empty/empty.vue [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            <slot v-if="$slots.description" name="description" />
+            <p v-else>{{ emptyDescription }}</p>
+        </div>
+        <div v-if="$slots.default" :class="`${prefixCls}-bottom`">
+            <slot />
+        </div>
+    </div>
+</template>
+
+<script lang="ts" setup>
```
Comment (wanchun) [inline components/empty/empty.vue:19]: 不用setup写法，不方便测试

File: components/empty/style/empty.less [hunk 3]
```diff
@@ -0,0 +1,60 @@
+@import '../../style/themes/index';
+@import '../../style/mixins/index';
+
+@empty-prefix-cls: ~'@{cls-prefix}-empty';
+
+@empty-padding: 30px 0;
+@empty-image-width: 150px;
+@empty-description-margin-top: 16px;
+@empty-bottom-margin-top: 24px;
+@empty-svg-monochrome-grey: #dcdfe6;
```
Comment (wanchun) [inline components/empty/style/empty.less:10]: 颜色从theme/base中取，不能随便定义颜色，会导致整个组件库没有风格

File: components/empty/style/empty.less [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+        img {
+            width: 100%;
+            height: 100%;
+            object-fit: contain;
+            vertical-align: top;
+            user-select: none;
+        }
+
+        svg {
```
Comment (wanchun) [inline components/empty/style/empty.less:39]: img和svg，属性选择器效率很低

File: components/empty/style/empty.less [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            height: 100%;
+            vertical-align: top;
+            fill: currentcolor;
+        }
+    }
+
+    &-description {
+        margin-top: var(--f-empty-description-margin-top);
+
+        p {
```
Comment (wanchun) [inline components/empty/style/empty.less:50]: 同上

File: components/empty/empty.vue [hunk 6]
```diff
@@ -0,0 +1,47 @@
+<template>
+    <div :class="prefixCls">
+        <div :class="`${prefixCls}-image`" :style="imageStyle">
+            <img v-if="imageSrc" :src="imageSrc" ondragstart="return false" />
+            <slot v-else name="image">
+                <DefaultImgEmpty />
+            </slot>
+        </div>
+        <div :class="`${prefixCls}-description`">
```
Comment (wanchun) [inline components/empty/empty.vue:9]: 没有description时，会渲染空的`<div :class="`${prefixCls}-description`">`，但是这层div有marginTop

File: components/empty/empty.vue [hunk 7]
```diff
@@ -0,0 +1,47 @@
+<template>
+    <div :class="prefixCls">
+        <div :class="`${prefixCls}-image`" :style="imageStyle">
+            <img v-if="imageSrc" :src="imageSrc" ondragstart="return false" />
```
Comment (wanchun) [inline components/empty/empty.vue:4]: 需在文档说明imageSrc优先级比插槽高

File: components/empty/empty.vue [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            <slot v-else name="image">
+                <DefaultImgEmpty />
+            </slot>
+        </div>
+        <div :class="`${prefixCls}-description`">
+            <slot v-if="$slots.description" name="description" />
+            <p v-else>{{ emptyDescription }}</p>
```
Comment (wanchun) [inline components/empty/empty.vue:11]: 这里插槽优先级又比emptyDescription低，一致性？

File: components/empty/style/empty.less [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    --f-empty-bottom-margin-top: @empty-bottom-margin-top;
+
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    box-sizing: border-box;
+    padding: var(--f-empty-padding);
+    text-align: center;
+
```
Comment (wanchun) [inline components/empty/style/empty.less:27]: 需加上.text()

File: components/empty/empty.vue [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+const props = defineProps(emptyProps);
+
+useTheme();
+
+const { t } = useLocale();
+const emptyDescription = computed(
+    () => props.description || t('empty.emptyText'),
+);
+
+const imageStyle = computed<CSSProperties>(() => props.imageStyle);
```
Comment (wanchun) [inline components/empty/empty.vue:46]: 直接用props.imageStyle

File: components/empty/empty.vue [hunk 11]
```diff
@@ -0,0 +1,47 @@
+<template>
+    <div :class="prefixCls">
+        <div :class="`${prefixCls}-image`" :style="imageStyle">
+            <img v-if="imageSrc" :src="imageSrc" ondragstart="return false" />
+            <slot v-else name="image">
+                <DefaultImgEmpty />
```
Comment (wanchun) [inline components/empty/empty.vue:6]: 直接用吧，组件会有性能损耗

File: components/empty/style/empty.less (hunk 3 above)
Comment (vimplus) [reply components/empty/style/empty.less]: 没用的，已删除

**Pull request merged**